src/main/java/com/itschool/eventmanagment/services/UserServiceImpl.java

```diff
@@ -3,22 +3,39 @@
 
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.itschool.eventmanagment.models.dtos.EventDTO;
+import com.itschool.eventmanagment.models.dtos.UserDTO;
+import com.itschool.eventmanagment.models.entities.Event;
+import com.itschool.eventmanagment.models.entities.User;
+import com.itschool.eventmanagment.repositories.EventRepository;
 import com.itschool.eventmanagment.repositories.UserRepository;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
+import java.util.Optional;
 
 @Service
 public class UserServiceImpl implements UserService {
     private final ObjectMapper objectMapper;
     private final UserRepository userRepository;
+    private final EventRepository eventRepository;
 
-    public UserServiceImpl(ObjectMapper objectMapper, UserRepository userRepository) {
+    public UserServiceImpl(ObjectMapper objectMapper, UserRepository userRepository, EventRepository eventRepository) {
         this.objectMapper = objectMapper;
         this.userRepository = userRepository;
+        this.eventRepository = eventRepository;
     }
 
-    public List<EventDTO> getUsersEvents() {
-        return null;
+    public List<EventDTO> getUserEvents(Long id) {
+        Optional<User> userWithEvent = userRepository.findById(id);
+        if(!userWithEvent.isPresent()) {
+            throw new RuntimeException("User with ID " + id + " not found");
+        }
+
+        User user = userWithEvent.get();
+        List<Event> events = eventRepository.findByUserId(user.getId());
+
+        return events.stream()
```

Comment:
I like the fact you use a functional implementation for this

Reply:
Thanks